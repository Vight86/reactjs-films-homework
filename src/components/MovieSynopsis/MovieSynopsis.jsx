import React from 'react';
import propTypes from 'prop-types';
import styles from './MovieSynopsis.scss';

const MovieSynopsis = ({ className, isShown, children }) => (
  <p className={`${styles[className]}${isShown ? ` ${styles.active}` : ''}`}>
    <span>
      {children}
    </span>
  </p>
);

MovieSynopsis.propTypes = {
  className: propTypes.string,
  isShown: propTypes.bool,
  children: propTypes.string,
};

MovieSynopsis.defaultProps = {
  className: 'primary',
  isShown: false,
  children: '',
};

export default MovieSynopsis;
