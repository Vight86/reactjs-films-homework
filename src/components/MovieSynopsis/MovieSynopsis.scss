@import '../../utils/variables';

.movieSynopsis {
  margin: 0;
  color: $color-light;
  font-size: 2.6em;
  line-height: 1.65;
}

.primary {
  @extend .movieSynopsis;

  position: absolute;
  right: 1em;
  bottom: 55%;
  max-width: 18em;
  box-sizing: border-box;
  padding: 1.15em;
  background-color: transparentize($color: #000, $amount: 0.36);
  border-radius: 3px;
  pointer-events: none;
  transform: scale(0);
  transform-origin: bottom center;
  transition: transform 0.3s ease-in;

  @media screen and (max-width: $bp-xxlarge) {
    bottom: 53%;
  }

  @media screen and (max-width: $bp-medium) {
    transform-origin: 78% bottom;
  }

  @media screen and (max-width: $bp-xxsmall) {
    top: -3em;
    right: 10em;
    left: 2em;
    max-width: none;
    transform-origin: right 25%;
  }

  & > span {
    display: block;
    overflow: hidden;
    width: 100%;
    height: 100%;
  }

  &::after {
    position: absolute;
    right: 50%;
    bottom: -1.05em;
    display: block;
    width: 1.9em;
    height: 1.1em;
    background: url('./arrowDown.svg') no-repeat center 100%;
    content: '';
    transform: translateX(-50%);

    @media screen and (max-width: $bp-xxlarge) {
      right: 10%;
    }

    @media screen and (max-width: $bp-xxsmall) {
      top: 26%;
      right: -1.5em;
      transform: rotate(-90deg);
    }

    @media screen and (max-width: $bp-xxxxsmall) {
      top: 30%;
    }
  }

  &.active {
    pointer-events: all;
    transform: scale(1);
    transition: transform 0.3s cubic-bezier(0.43, 1.95, 0.66, 0.63);
  }
}

.secondary_grid {
  @extend .movieSynopsis;

  position: relative;
  z-index: 10;
  overflow: hidden;
  max-height: 8.1em;
  padding: 0 8% 0 8%;
  margin: 9.5% 0 10.7% 0;

  @media screen and (max-width: $bp-xlarge) {
    margin: 0 0 8% 0;
  }
}

.secondary_table {
  @extend .secondary_grid;

  margin: 70px 0 40px;

  @media screen and (max-width: $bp-xlarge) {
    margin: 20px 0;
  }
}

.tertiary {
  @extend .movieSynopsis;

  overflow: hidden;
  max-height: 8.1em;
  margin-top: 2em;
  color: $text-tertiary;
  font-size: 2em;
}
