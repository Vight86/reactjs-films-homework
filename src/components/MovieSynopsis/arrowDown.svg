<svg xmlns="http://www.w3.org/2000/svg" width="48.71" height="28.156" viewBox="0 0 48.71 28.156">
  <defs>
    <style>
      .cls-1 {
        fill-rule: evenodd;
        opacity: 0.64;
      }
    </style>
  </defs>
  <path class="cls-1" d="M2168.13,967.777l21.3-21.886a3.05,3.05,0,0,0,.9-2.2,3.017,3.017,0,0,0-.91-2.2,2.876,2.876,0,0,0-2.14-.927l-42.62.022a2.89,2.89,0,0,0-2.14.929,3.132,3.132,0,0,0,0,4.394l21.33,21.865a2.875,2.875,0,0,0,2.14.926,2.906,2.906,0,0,0,2.14-.929h0Z" transform="translate(-2141.63 -940.563)"/>
</svg>
