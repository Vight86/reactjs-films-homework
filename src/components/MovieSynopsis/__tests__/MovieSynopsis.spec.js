import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MovieSynopsis from '../index';

const setUp = (className, isShown, children) => {
  const renderer = new ShallowRenderer();
  renderer.render(
    <MovieSynopsis className={className} isShown={isShown}>{children}</MovieSynopsis>,
  );
  return renderer.getRenderOutput();
};

describe('render MovieList component', () => {
  it('render component without props', () => {
    const result = setUp();
    expect(result).toMatchSnapshot();
  });

  it('render component with props', () => {
    const result = setUp('class', true, 'children');
    expect(result).toMatchSnapshot();
  });
});
