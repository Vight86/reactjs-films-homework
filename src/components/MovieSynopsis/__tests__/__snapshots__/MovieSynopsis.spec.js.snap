// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MovieList component render component with props 1`] = `
<p
  className="class active"
>
  <span>
    children
  </span>
</p>
`;

exports[`render MovieList component render component without props 1`] = `
<p
  className="primary"
>
  <span>
    
  </span>
</p>
`;
